FluffOS v2.26.1 CYGWIN Compile failure

$ yacc -V
yacc - 1.9 20121003

$ bison -V
bison (GNU Bison) 2.7
Written by Robert Corbett and Richard Stallman.
GCC 4.7.2
Code: [Select]
$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/i686-pc-cygwin/4.7.2/lto-wrapper.exe
Target: i686-pc-cygwin
Configured with: /cygdrive/i/szsz/tmpp/gcc4/gcc4-4.7.2-1/src/gcc-4.7.2/configure
 --srcdir=/cygdrive/i/szsz/tmpp/gcc4/gcc4-4.7.2-1/src/gcc-4.7.2 --prefix=/usr --
exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --libexecdir=/usr/lib --d
atadir=/usr/share --localstatedir=/var --sysconfdir=/etc --datarootdir=/usr/shar
e --docdir=/usr/share/doc/gcc4 -C --datadir=/usr/share --infodir=/usr/share/info
 --mandir=/usr/share/man -v --with-gmp=/usr --with-mpfr=/usr --enable-bootstrap
--enable-version-specific-runtime-libs --libexecdir=/usr/lib --enable-static --e
nable-shared --enable-shared-libgcc --disable-__cxa_atexit --with-gnu-ld --with-
gnu-as --with-dwarf2 --disable-sjlj-exceptions --enable-languages=ada,c,c++,fort
ran,lto,objc,obj-c++ --enable-graphite --enable-lto --enable-java-awt=gtk --disa
ble-symvers --enable-libjava --program-suffix=-4 --enable-libgomp --enable-libss
p --enable-libada --enable-threads=posix --with-arch=i686 --with-tune=i686 --ena
ble-libgcj-sublibs CC=gcc-4 CXX=g++-4 CC_FOR_TARGET=gcc-4 CXX_FOR_TARGET=g++-4 G
NATMAKE_FOR_TARGET=gnatmake GNATBIND_FOR_TARGET=gnatbind --with-ecj-jar=/usr/sha
re/java/ecj.jar

$ uname -a
CYGWIN_NT-6.1 wireless 1.7.17(0.262/5/3) 2012-10-19 14:39 i686 Cygwin

```
Andy@wireless /cygdrive/c/mud/fluffos-2.26.1
$ ./build.FluffOS
Preparing to build standard MudOS driver ...
Trying out some stuff to see what works; ignore errors ...
gmake: Nothing to be done for `nothing'.
Using standard a.out for compiler default executable
./build.FluffOS: line 197: xlc: command not found
comptest.c:1:0: sorry, unimplemented: 64-bit mode not compiled in
mkdir: cannot create directory `tmp': File exists
install: unknown option -- f
Try `install --help' for more information.
rm: cannot remove `tmp/insttest': No such file or directory
./build.FluffOS: line 379: ./a.out: No such file or directory
rm: cannot remove `a.out': No such file or directory
***************** Configuration completed **************
Installing MudOS on

Using install -c to install binaries in ../bin.
Using gcc -std=c99 -D_GNU_SOURCE -flto -E  for preprocessing.
Using gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__            -O3
 to compile.
Using bison -d -y  to make the compiler.
Edit GNUmakefile if this is not what you want

Otherwise, type 'gmake' to build MudOS, then 'gmake install'.

Andy@wireless /cygdrive/c/mud/fluffos-2.26.1
$ gmake
rm -f cc.h
echo "/* this file automatically generated by the Makefile */" > cc.h
echo '#define COMPILER "gcc -std=c99 -D_GNU_SOURCE -flto"' >> cc.h
echo '#define OPTIMIZE "-O3"' >> cc.h
echo '#define CFLAGS   "-D__USE_FIXED_PROTOTYPES__ -O3"' >> cc.h
echo '#define OBJDIR   "obj"' >> cc.h
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -o obj/edit_source.o
 -c edit_source.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/hash.o -c
 hash.c
bison -d -y  make_func.y
rm -f make_func.tab.c
mv y.tab.c make_func.tab.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -o obj/make_func.tab
.o -c make_func.tab.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ obj/edit_source.o ob
j/hash.o obj/make_func.tab.o -o edit_source
if test \( ! -r configure.h \) -o \( ! -r configuration \); then \
    rm -f configuration; \
    touch configuration; \
fi
if test "Machine `uname -a` Configure version 5" = "`cat configuration`"; then \

    echo "Skipping configuration ..."; \
else \
    ./edit_source -configure; \
    echo "Machine `uname -a` Configure version 5" > configuration; \
fi
Checking for include file <stdlib.h> ... exists
Checking for include file <unistd.h> ... exists
Checking for include file <time.h> ... exists
Checking for include file <sys/times.h> ... exists
Checking for include file <fcntl.h> ... exists
Checking for include file <sys/time.h> ... exists
Checking for include file <dos.h> ... does not exist
Checking for include file <usclkc.h> ... does not exist
Checking for include file <limits.h> ... exists
Checking for include file <locale.h> ... exists
Checking for include file <netinet/in.h> ... exists
Checking for include file <arpa/inet.h> ... exists
Checking for include file <sys/types.h> ... exists
Checking for include file <sys/ioctl.h> ... exists
Checking for include file <sys/socket.h> ... exists
Checking for include file <netdb.h> ... exists
Checking for include file <arpa/telnet.h> ... exists
Checking for include file <sys/sema.h> ... does not exist
Checking for include file <sys/socketvar.h> ... does not exist
Checking for include file <socket.h> ... does not exist
Checking for include file <resolve.h> ... does not exist
Checking for include file <sys/stat.h> ... exists
Checking for include file <dirent.h> ... exists
Checking for include file <sys/dirent.h> ... exists
Checking for include file <sys/filio.h> ... does not exist
Checking for include file <sys/sockio.h> ... does not exist
Checking for include file <sys/mkdev.h> ... does not exist
Checking for include file <sys/resource.h> ... exists
Checking for include file <sys/rusage.h> ... does not exist
Checking for include file <sys/wait.h> ... exists
Checking for include file <sys/crypt.h> ... does not exist
Checking for include file <crypt.h> ... exists
Checking for include file <my_malloc.h> ... does not exist
Checking for include file <mach/mach.h> ... does not exist
Checking for include file <mach.h> ... does not exist
Checking for include file <sys/sysmacros.h> ... exists
Checking for include file <dlfcn.h> ... exists
Checking for missing INADDR_NONE ... ok
Checking for random number generator ... using drand48()
Using BSD signals.
Checking if signal() returns SIG_ERR on error ... yes
Not Checking for inline ...(usage in driver code all broken anyway) const ...
Checking for strerror() ... exists
Checking for POSIX getcwd() ... exists
Checking for getrusage() ... exists
Checking for times() ... exists
Checking for gettimeofday() ... exists
Checking for fchmod() ... exists
Checking for big or little endian ... little
Checking for memmove() ... exists
Checking for library -lresolv ... exists
Checking for library -lbsd ... does not exist
Checking for library -lBSD ... does not exist
Checking for library -ly ... exists
Checking for library -lcrypt ... exists
Checking for library -lsocket ... does not exist
Checking for library -linet ... does not exist
Checking for library -lnsl ... does not exist
Checking for library -lnsl_s ... does not exist
Checking for library -lseq ... does not exist
Checking for library -lm ... exists
Checking for library -lz ... exists
Checking for library -lpthread ... exists
Checking for flaky Linux systems ...

./edit_source -options -malloc -build_func_spec 'gcc -std=c99 -D_GNU_SOURCE -flt
o -E -D__USE_FIXED_PROTOTYPES__' \
              -process grammar.y.pre
No "local_options" file present.  If you create one from "options.h",
you can use it when you get a new driver, and you will be warned if there are
changes to the real options.h which you should include in your local file.
Writing build options to option_defs.c ...
Using memory allocation package: sysmalloc.c
Building compiler files ...
Creating 'grammar.y' from 'grammar.y.pre' ...

./edit_source -process packages/Makefile.pre
Creating 'packages/Makefile' from 'packages/Makefile.pre' ...

./edit_source -process packages/GNUmakefile.pre
Creating 'packages/GNUmakefile' from 'packages/GNUmakefile.pre' ...

./edit_source -build_efuns -build_applies
Building efun tables ...

touch mallocwrapper.c
touch malloc.c
touch files
gmake main_build2
gmake[1]: Entering directory `/cygdrive/c/mud/fluffos-2.26.1'
gmake -C packages 'CC=gcc -std=c99 -D_GNU_SOURCE -flto' 'CFLAGS=-D__USE_FIXED_PR
OTOTYPES__ -O3' 'OBJDIR=../obj' 'RANLIB=ranlib' 'A=a' 'O=o'
gmake[2]: Entering directory `/cygdrive/c/mud/fluffos-2.26.1/packages'
gcc -std=c99 -D_GNU_SOURCE -flto -I../obj -I.. -D__USE_FIXED_PROTOTYPES__ -O3 -o
 ../obj/uids.o -c uids.c
gcc -std=c99 -D_GNU_SOURCE -flto -I../obj -I.. -D__USE_FIXED_PROTOTYPES__ -O3 -o
 ../obj/compress.o -c compress.c
gcc -std=c99 -D_GNU_SOURCE -flto -I../obj -I.. -D__USE_FIXED_PROTOTYPES__ -O3 -o
 ../obj/math.o -c math.c
gcc -std=c99 -D_GNU_SOURCE -flto -I../obj -I.. -D__USE_FIXED_PROTOTYPES__ -O3 -o
 ../obj/develop.o -c develop.c
gcc -std=c99 -D_GNU_SOURCE -flto -I../obj -I.. -D__USE_FIXED_PROTOTYPES__ -O3 -o
 ../obj/external.o -c external.c
gcc -std=c99 -D_GNU_SOURCE -flto -I../obj -I.. -D__USE_FIXED_PROTOTYPES__ -O3 -o
 ../obj/contrib.o -c contrib.c
contrib.c: In function `f_terminal_colour':
contrib.c:680:4: warning: implicit declaration of function `alloca' [-Wimplicit-
function-declaration]
contrib.c: In function `pluralize':
contrib.c:1067:3: warning: implicit declaration of function `strcasecmp' [-Wimpl
icit-function-declaration]
contrib.c: In function `set_timezone':
contrib.c:1771:2: warning: implicit declaration of function `putenv' [-Wimplicit
-function-declaration]
contrib.c:1772:2: warning: implicit declaration of function `tzset' [-Wimplicit-
function-declaration]
contrib.c: In function `reset_timezone':
contrib.c:1784:3: warning: implicit declaration of function `unsetenv' [-Wimplic
it-function-declaration]
gcc -std=c99 -D_GNU_SOURCE -flto -I../obj -I.. -D__USE_FIXED_PROTOTYPES__ -O3 -o
 ../obj/sockets.o -c sockets.c
gcc -std=c99 -D_GNU_SOURCE -flto -I../obj -I.. -D__USE_FIXED_PROTOTYPES__ -O3 -o
 ../obj/async.o -c async.c
gcc -std=c99 -D_GNU_SOURCE -flto -I../obj -I.. -D__USE_FIXED_PROTOTYPES__ -O3 -o
 ../obj/dwlib.o -c dwlib.c
dwlib.c: In function `f_add_a':
dwlib.c:561:2: warning: implicit declaration of function `strncasecmp' [-Wimplic
it-function-declaration]
ar rcu packages.a ../obj/uids.o ../obj/compress.o ../obj/math.o ../obj/develop.o
 ../obj/external.o ../obj/contrib.o ../obj/sockets.o ../obj/async.o ../obj/dwlib
.o
ranlib packages.a
gmake[2]: Leaving directory `/cygdrive/c/mud/fluffos-2.26.1/packages'
bison -d -y grammar.y
rm -f grammar.tab.*
sed "s/y.tab.c/grammar.tab.c/g" y.tab.c  > grammar.tab.c
mv y.tab.h grammar.tab.h
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/grammar.t
ab.o -c grammar.tab.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/lex.o -c
lex.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/main.o -c
 main.c
main.c: In function `main':
main.c:112:2: warning: implicit declaration of function `tzset' [-Wimplicit-func
tion-declaration]
main.c:163:2: warning: implicit declaration of function `srand48' [-Wimplicit-fu
nction-declaration]
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/rc.o -c r
c.c
rc.c: In function `set_defaults':
rc.c:220:3: warning: implicit declaration of function `strcasecmp' [-Wimplicit-f
unction-declaration]
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/interpret
.o -c interpret.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/simulate.
o -c simulate.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/file.o -c
 file.c
file.c: In function `read_bytes':
file.c:566:5: warning: implicit declaration of function `fileno' [-Wimplicit-fun
ction-declaration]
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/object.o
-c object.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/backend.o
 -c backend.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/array.o -
c array.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/mapping.o
 -c mapping.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/comm.o -c
 comm.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/ed.o -c e
d.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/regexp.o
-c regexp.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/buffer.o
-c buffer.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/crc32.o -
c crc32.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/malloc.o
-c malloc.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/mallocwra
pper.o -c mallocwrapper.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/class.o -
c class.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/efuns_mai
n.o -c efuns_main.c
gcc -std=c99 -D_GNU_SOURCE -flto -D__USE_FIXED_PROTOTYPES__ -O3 -o obj/efuns_por
t.o -c efuns_port.c
efuns_port.c: In function `f_localtime':
efuns_port.c:146:41: error: `timezone' undeclared (first use in this function)
efuns_port.c:146:41: note: each undeclared identifier is reported only once for
each function it appears in
efuns_port.c:147:39: error: `tzname' undeclared (first use in this function)
GNUmakefile:91: recipe for target `obj/efuns_port.o' failed
gmake[1]: *** [obj/efuns_port.o] Error 1
gmake[1]: Leaving directory `/cygdrive/c/mud/fluffos-2.26.1'
GNUmakefile:84: recipe for target `main_build' failed
gmake: *** [main_build] Error 2

Andy@wireless /cygdrive/c/mud/fluffos-2.26.1
```

The problem is 

```
efuns_port.c: In function `f_localtime':
efuns_port.c:146:41: error: `timezone' undeclared (first use in this function)
efuns_port.c:146:41: note: each undeclared identifier is reported only once for
each function it appears in
efuns_port.c:147:39: error: `tzname' undeclared (first use in this function)
GNUmakefile:91: recipe for target `obj/efuns_port.o' failed
gmake[1]: *** [obj/efuns_port.o] Error 1
```

--- a/src/edit_source.c
+++ b/src/edit_source.c
@@ -1457,6 +1457,7 @@ static void handle_configure() {
     check_include("INCL_SYS_CRYPT_H", "sys/crypt.h");
     check_include("INCL_CRYPT_H", "crypt.h");
     check_include("INCL_MALLOC_H", "my_malloc.h");
+    check_include("INCL_ALLOCA_H", "alloca.h");
 
     /* for NeXT */
     if (!check_include("INCL_MACH_MACH_H", "mach/mach.h"))
@@ -1514,13 +1515,13 @@ static void handle_configure() {
         printf(" no\n");
     }
 
-    printf("Not Checking for inline ...(usage in driver code all broken anyway)");
-    //if (!check_prog("INLINE inline", "inline void foo() { }", "foo();", 0)) {
-        //printf(" __inline ...");
-        //if (!check_prog("INLINE __inline", "__inline void foo() {}", "foo();", 0)) {
+    printf("Checking for inline ...");
+    if (!check_prog("INLINE inline", "inline void foo() { }", "foo();", 0)) {
+        printf(" __inline ...");
+        if (!check_prog("INLINE __inline", "__inline void foo() {}", "foo();", 0)) {
             fprintf(yyout, "#define INLINE\n");
-        //}
-    //}
+        }
+    }
     printf(" const ...\n");
     if (!check_prog("CONST const", "int foo(const int *, const int *);", "", 0))
         fprintf(yyout, "#define CONST\n");
@@ -1662,6 +1663,8 @@ static void handle_configure() {
         check_library("-lcrypto");
     if (lookup_define("POSIX_TIMERS"))
         check_library("-lrt");
+    if (lookup_define("USE_ICONV"))
+        check_library("-liconv");
     fprintf(stderr, "Checking for flaky Linux systems ...\n");
     check_linux_libc();
 

--- a/src/efuns_port.c
+++ b/src/efuns_port.c
@@ -22,6 +22,14 @@
 #include <limits.h>
 #endif
 
+/* work around mingw32/cygwin32 c99 bug.*/
+#ifdef __CYGWIN__
+/* defines for the opengroup specifications Derived from Issue 1 of the SVID.  */
+extern __IMPORT long _timezone;
+extern __IMPORT int _daylight;
+extern __IMPORT char *_tzname[2];
+#endif
+
 #ifdef F_CRYPT
 #define SALT_LEN        8
 #ifdef CUSTOM_CRYPT
@@ -140,14 +148,12 @@ f_localtime (void)
 #else                           /* sequent */
 #if (defined(hpux) || defined(_SEQUENT_) || defined(_AIX) || defined(SunOS_5) \
     || defined(SVR4) || defined(sgi) || defined(__linux__) || defined(cray) \
-    || defined(__CYGWIN__)\
     )
       if (!tm->tm_isdst) {
         vec->item[LT_GMTOFF].u.number = timezone;
         vec->item[LT_ZONE].u.string = string_copy(tzname[0], "f_localtime");
       } else {
 #if (defined(_AIX) || defined(hpux) || defined(__linux__) || defined(cray) \
-    || defined(__CYGWIN__)\
     )
         vec->item[LT_GMTOFF].u.number = timezone;
 #else
@@ -156,13 +162,13 @@ f_localtime (void)
         vec->item[LT_ZONE].u.string = string_copy(tzname[1], "f_localtime");
       }
 #else
-#ifndef WIN32
-      vec->item[LT_GMTOFF].u.number = tm->tm_gmtoff;
-      vec->item[LT_ZONE].u.string = string_copy(tm->tm_zone, "f_localtime");
-#else
+#if defined(WIN32) || defined(__CYGWIN32__)
       vec->item[LT_GMTOFF].u.number = _timezone;
       vec->item[LT_ZONE].u.string = string_copy(_tzname[_daylight?1:0],"f_localtime");
-#endif
+#else
+      vec->item[LT_GMTOFF].u.number = tm->tm_gmtoff;
+      vec->item[LT_ZONE].u.string = string_copy(tm->tm_zone, "f_localtime");
+#endif                          /* win32 | cygwin32 */
 #endif
 #endif                          /* sequent */
 #endif                          /* BSD42 */

--- a/src/packages/contrib.c
+++ b/src/packages/contrib.c
@@ -1,4 +1,5 @@
 #define SUPPRESS_COMPILER_INLINES
+#include "../std.h"
 #include "../lpc_incl.h"
 #include "../comm.h"
 #include "../file_incl.h"

--- a/src/std_incl.h
+++ b/src/std_incl.h
@@ -105,7 +105,7 @@ int dos_style_link (char *, char *);
 #  include <mach.h>
 #endif
 
-#ifdef sun
+#ifdef INCL_ALLOCA_H
 #  include <alloca.h>
 #endif
 

